stdlib/public/core/StringBridge.swift

```diff
@@ -55,68 +134,100 @@ internal func _cocoaStringCopyCharacters(
   range: Range<Int>,
   into destination: UnsafeMutablePointer<UTF16.CodeUnit>
 ) {
-  _swift_stdlib_CFStringGetCharacters(
-    source,
-    _swift_shims_CFRange(location: range.lowerBound, length: range.count),
-    destination)
+  _NSStringGetCharacters(from: _objc(source), range: range, into: destination)
+}
+
+@_effects(readonly)
+private func _NSStringGetCharacter(
```

Comment:
Similarly with suggestion about `_copyNSString`, perhaps this body should get inlined into `_cocoaStringSubscript`.